File: android/src/main/java/com/bixolon/bxlflutterbgatelib/BxlflutterbgatelibPlugin.java

package com.bixolon.bxlflutterbgatelib;
import android.app.Activity;
import androidx.annotation.NonNull;
import io.flutter.embedding.engine.plugins.FlutterPlugin;
import io.flutter.embedding.engine.plugins.activity.ActivityAware;
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
import io.flutter.plugin.common.EventChannel;
import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;

/** BxlflutterbgatelibPlugin */
public class BxlflutterbgatelibPlugin implements FlutterPlugin, MethodCallHandler, ActivityAware {

  private static Activity activity;

  @Override
  public void onAttachedToEngine(@NonNull FlutterPluginBinding flutterPluginBinding) {
    new BxlMPosControllerBCDPlugin().onAttachedToEngine(flutterPluginBinding);
    new BxlMPosControllerConfigPlugin().onAttachedToEngine(flutterPluginBinding);
    new BxlMPosControllerDallasKeyPlugin().onAttachedToEngine(flutterPluginBinding);
    new BxlMPosControllerHIDPlugin().onAttachedToEngine(flutterPluginBinding);
    new BxlMPosControllerLabelPrinterPlugin().onAttachedToEngine(flutterPluginBinding);
    new BxlMPosControllerMSRPlugin().onAttachedToEngine(flutterPluginBinding);
    new BxlMPosControllerNFCPlugin().onAttachedToEngine(flutterPluginBinding);
    new BxlMPosControllerPrinterPlugin().onAttachedToEngine(flutterPluginBinding);
    new BxlMPosControllerRFIDPlugin().onAttachedToEngine(flutterPluginBinding);
    new BxlMPosControllerScalePlugin().onAttachedToEngine(flutterPluginBinding);
    new BxlMPosControllerScannerPlugin().onAttachedToEngine(flutterPluginBinding);
    new BxlMPosControllerUSBToSerialPlugin().onAttachedToEngine(flutterPluginBinding);
    new BxlMPosLookupPlugin().onAttachedToEngine(flutterPluginBinding);
  }
  @Override
  public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) { }
  @Override
  public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) { }
  @Override
  public void onAttachedToActivity(@NonNull ActivityPluginBinding binding) {
    BxlflutterbgatelibPlugin.activity = binding.getActivity();
  }
  @Override
  public void onReattachedToActivityForConfigChanges(@NonNull ActivityPluginBinding binding) {
    BxlflutterbgatelibPlugin.activity = binding.getActivity();
  }
  @Override
  public void onDetachedFromActivity() {
    BxlflutterbgatelibPlugin.activity = null;
  }
  @Override
  public void onDetachedFromActivityForConfigChanges() {
    BxlflutterbgatelibPlugin.activity = null;
  }

  public static void runInUiThread(EventChannel.EventSink eventSink, Integer param){
    if(BxlflutterbgatelibPlugin.activity == null)
      return;
    if(eventSink == null)
      return;
    BxlflutterbgatelibPlugin.activity.runOnUiThread(() -> {
      try {
        eventSink.success(param);
      } catch (Exception e) {
        e.printStackTrace();
      }
    });
  }
}

File: ios/Classes/BxlflutterbgatelibPlugin.m
@import frameworkMPosSDK;

#import "BxlflutterbgatelibPlugin.h"
#import "BxlMPosLibVersionPlugin.h"
#import "BxlMPosLookupPlugin.h"
#import "BxlMPosControllerDevicePlugin.h"
#import "BxlMPosControllerConfigPlugin.h"
#import "BxlMPosControllerPrinterPlugin.h"
#import "BxlMPosControllerLabelPrinterPlugin.h"
#import "BxlMPosControllerBcdPlugin.h"
#import "BxlMPosControllerDallasKeyPlugin.h"
#import "BxlMPosControllerHidPlugin.h"
#import "BxlMPosControllerNfcPlugin.h"
#import "BxlMPosControllerRfidPlugin.h"
#import "BxlMPosControllerScalePlugin.h"
#import "BxlMPosControllerScannerPlugin.h"
#import "BxlMPosControllerTtyUsbPlugin.h"


@implementation BxlflutterbgatelibPlugin

+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
    [BxlMPosLibVersionPlugin registerWithRegistrar:registrar];
    [BxlMPosLookupPlugin registerWithRegistrar:registrar];
    [BxlMPosControllerDevicePlugin registerWithRegistrar:registrar];
    [BxlMPosControllerConfigPlugin registerWithRegistrar:registrar];
    [BxlMPosControllerPrinterPlugin registerWithRegistrar:registrar];
    [BxlMPosControllerLabelPrinterPlugin registerWithRegistrar:registrar];
    [BxlMPosControllerBcdPlugin registerWithRegistrar:registrar];
    [BxlMPosControllerDallasKeyPlugin registerWithRegistrar:registrar];
    [BxlMPosControllerHidPlugin registerWithRegistrar:registrar];
    [BxlMPosControllerNfcPlugin registerWithRegistrar:registrar];
    [BxlMPosControllerRfidPlugin registerWithRegistrar:registrar];
    [BxlMPosControllerScalePlugin registerWithRegistrar:registrar];
    [BxlMPosControllerScannerPlugin registerWithRegistrar:registrar];
    [BxlMPosControllerTtyUsbPlugin registerWithRegistrar:registrar];
}

@end
